### PR TITLE
add publisher direct methods doc

### PR DESCRIPTION
**Pull request by cristipogacean**
add publisher direct methods documentation chapters

### CODE REVIEW

**Review by karok2m — APPROVED** [review]

File: docs/modules/publisher.md [hunk 1]
```diff
@@ -121,24 +126,205 @@ A basic configuration file looks like this:
...
+
+Consists in the following methods that are inherited from 2.5.x plus some additional ones
+
+- PublishNodes_V1
+- UnpublishNodes_V1
+- UnpublishAllNodes_V1
+- GetConfiguredEndpoints_V1
+- GetConfiguredNodesOnEndpoints_V1
+- GetDiagnosticInfo_V1
+- AddOrUpdateNodes_v1
```
Comment (karok2m) [inline docs/modules/publisher.md]: Please rename `AddOrUpdateNodes_v1` to `AddOrUpdateEndpoints_V1`. We had a discussion about this during one of the review sessions and as far as I remember we decided that `AddOrUpdateEndpoints` will be a more descriptive name as we are changing endpoints here. I have just updated the spec with this new name.

File: docs/modules/publisher.md [hunk 2]
```diff
@@ -121,24 +126,205 @@ A basic configuration file looks like this:
...
+| `DataSetPublishingInterval`      | no        | Integer | false                     | The publishing interval used for a grouped set of nodes <br>under a certain DataSetWriter. When defined it shall <br>override the OpcPublishingInterval value in the OpcNodes <br>if grouped underneath a DataSetWriter. |
+| `Tag`                            | no        | String  | empty                     | TODO                                                                                         |
+| `OpcNodes`                       | yes*      | OpcNode | empty                     | The DataSet collection grouping the nodes to be published for <br>the specific DataSetWriter defined above. |
+
+*Note*: *OpcNodes field is mandatory for PublishNodes_V1, UnpublishNodes_V1 and AddOrUpdateNodes_V1. It shall not be specified for the rest of the methods.
```
Comment (karok2m) [inline docs/modules/publisher.md]: Please rename `AddOrUpdateNodes_v1` to `AddOrUpdateEndpoints_V1`.

File: docs/modules/publisher.md [hunk 3]
```diff
@@ -121,24 +126,205 @@ A basic configuration file looks like this:
...
+
+  *Request:*
+
+  *Response:*
+
+  *Exceptions:*
+
+  *Example:*
+
+#### **AddOrUpdateNodes_v1**
```
Comment (karok2m) [inline docs/modules/publisher.md]: Please rename `AddOrUpdateNodes_v1` to `AddOrUpdateEndpoints_V1`.

File: docs/modules/publisher.md (hunk 2 above)
Comment (karok2m) [reply docs/modules/publisher.md:212]: Also, what is `*` in front of `OpcNodes` intended for ? Seems like it is missing a closing `*` if you had italic in mind, or double `*`s if you wanted it to be bold.

Comment (karok2m) [reply docs/modules/publisher.md:212]: Ah, ok, got it, `*` is an explanation for `yes*` above. Please ignore the comment about `*`

File: docs/modules/publisher.md [hunk 4]
```diff
@@ -121,24 +126,205 @@ A basic configuration file looks like this:
...
+| `HeartbeatIntervalTimespan`      | No        | String  | null                      | The interval used for the node to publish a value (a publisher <br>cached one) even if the value has not been <br>changed at the source. This value is represented in seconds. <br>Value expressed in Timespan string({d.hh:mm:dd.fff}).     |
+| `SkipFirst`                      | No        | Boolean | false                     | Instructs the publisher not to add to telemetry the<br> Initial DataChange (after subscription activation) for this OpcNode.     |
+| `QueueSize`                      | No        | Integer | 1                         | The desired QueueSize for the monitored item to be published.     |
+
+*Note*: **Id field may be omitted when ExpandedNodeIdId is present.
+
+A DataSet is a group of NodeIds within an OPC UA Server to be published with a same publishing interval.  
+DataSetWriter has one DataSet and contains the elements required to successfully establish a connection to the OPC UA Server. 
+A DatSetGroup is used to several DataSetWriters for a specific OPC UA server.
```
Comment (karok2m) [inline docs/modules/publisher.md]: Is there a verb missing after `used to` ? Should it have been `used to group` ?

File: docs/modules/publisher.md [hunk 5]
```diff
@@ -121,24 +126,205 @@ A basic configuration file looks like this:
...
+
+*Note*: **Id field may be omitted when ExpandedNodeIdId is present.
+
+A DataSet is a group of NodeIds within an OPC UA Server to be published with a same publishing interval.  
+DataSetWriter has one DataSet and contains the elements required to successfully establish a connection to the OPC UA Server. 
+A DatSetGroup is used to several DataSetWriters for a specific OPC UA server. 
+
+#### **PublishNodes_V1**
+
+PublishNodes enables a client to add a set of nodes to be published for a specific DataSetWriter. When a DataSetWriter already exist, the nodes shall be incrementally added to the very same dataset. When it does not already exist, a new DataSetWritter shall be created with the initial set of nodes contained in the request.
```
Comment (karok2m) [inline docs/modules/publisher.md]: I believe that there is `s` missing after `DataSetWriter already exist`. Looks like it should be `exists`.

File: docs/modules/publisher.md (hunk 1 above)
Comment (cristipogacean) [reply docs/modules/publisher.md]: right

File: docs/modules/publisher.md (hunk 2 above)
Comment (cristipogacean) [reply docs/modules/publisher.md]: right

File: docs/modules/publisher.md (hunk 3 above)
Comment (cristipogacean) [reply docs/modules/publisher.md]: done

File: docs/modules/publisher.md (hunk 4 above)
Comment (cristipogacean) [reply docs/modules/publisher.md]: done

File: docs/modules/publisher.md (hunk 5 above)
Comment (cristipogacean) [reply docs/modules/publisher.md]: right

**Review by karok2m — APPROVED** [review]

**Pull request merged**